src/components/views/elements/CustomInput.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useRef, useEffect } from 'react';
+
+// via https://itnext.io/reusing-the-ref-from-forwardref-with-react-hooks-4ce9df693dd
+function useCombinedRefs(...refs) {
+    const targetRef = useRef();
+
+    useEffect(() => {
+        refs.forEach(ref => {
+            if (!ref) return;
+
+            if (typeof ref === 'function') {
+                ref(targetRef.current);
+            } else {
+                ref.current = targetRef.current;
+            }
+        });
+    }, [refs]);
+
+    return targetRef;
+}
+
+interface CustomInputProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange' | 'onInput'> {
+    onChange?: (event: Event) => void;
+    onInput?: (event: Event) => void;
+}
+/**
+* This component restores the native 'onChange' and 'onInput' behavior of
+* JavaScript. via https://stackoverflow.com/a/62383569/796832 and
+* https://github.com/facebook/react/issues/9657#issuecomment-643970199
+*
+* See:
+* - https://reactjs.org/docs/dom-elements.html#onchange
+* - https://github.com/facebook/react/issues/3964
+* - https://github.com/facebook/react/issues/9657
+* - https://github.com/facebook/react/issues/14857
+*
+* We use this for the <input type="date"> date picker so we can distinguish
+* from a final date picker selection vs navigating the months in the date
+* picker which trigger an `input`(and `onChange` in React).
+*/
+const CustomInput: React.FC<CustomInputProps> = React.forwardRef((props: CustomInputProps, ref) => {
```

Comment:
my major concern with this component is that it'll be the 5th of its kind in the project. We have `AccessibleButton`, `StyledRadioButton`, `StyledCheckbox`, `Field`, etc which all override some minor bit of behaviour that is slightly different from what React wants us to do. 

Can we instead mix this into `Field` or something to avoid having another surface to worry about?

Reply:
What weird non-React things are those components doing or getting around? Those seem pretty fine to me from glancing through.

 - `AccessibleButton` makes any element as accessible as a `<button>`
 - `StyledRadioButton` and `StyledCheckbox` appear to just add some more markup and styling around radiuos/checkboxes
 - `Field` the standard styled `<input>` for Element. Pretty involved but it's still a standard, known-quantity throughout the app.

I feel like `CustomInput` is the weirdest of them all and is much more exotic than the existing components (it actually works around React). Very hacky but `CustomInput` encapsulates the weirdness pretty well. 🤞 hoping some React solution can make it obsolete!

> Can we instead mix this into `Field` or something to avoid having another surface to worry about?

I think we could mix it in but it kinda sucks to have all of that non-standard stuff in the `Field` which is reasoned about often. The `onchange` vs `oninput` difference is only relevant to a few `<input>` types and are used rarely.

Feel free to lean harder in this direction to make the change.

Reply:
can we call it `OnChangeableInput`, `NativeChangeInput`, or something like that which describes its purpose? Naming it is quite difficult, which sort of indicates to me it's not really a component in its own right, but if it can self document then I'll be happy enough about it :p 

Unfortunately "custom" describes the vast majority of the project, so would like to give it a proper name.

Reply:
Great shout! Blinded by the copy-paste there -> `NativeOnChangeInput` ⏩